src/DocumentStore.zig

```diff
@@ -133,6 +142,21 @@ fn loadBuildAssociatedConfiguration(allocator: std.mem.Allocator, build_file: *B
         defer allocator.free(absolute_builtin_path);
         build_file.builtin_uri = try URI.fromPath(allocator, absolute_builtin_path);
     }
+
+    if (build_associated_config.build_options) |opts| {
+        // Copy options out of json parse result since they will be invalidated
+        var build_opts = try std.ArrayListUnmanaged(BuildAssociatedConfig.BuildOption).initCapacity(allocator, opts.len);
+        errdefer {
+            for (build_opts.items) |*opt| {
+                opt.deinit(allocator);
+            }
+            build_opts.deinit(allocator);
+        }
+        for (opts) |opt| {
+            try build_opts.append(allocator, try opt.dupe(allocator));
+        }
+        build_file.build_options = build_opts.toOwnedSlice(allocator);
+    }
```

Comment:
You can probably do something like  
```zig
build_file.build_options = build_associated_config.build_options;
build_associated_config.build_options = null;
```  
To 'steal' the slice from the parsed JSON since the allocators are the same.  
In any case using an ArrayList is overkill since you can also allocate a slice of the desired size with Allocator.alloc